Add segment analytics pump

JosephEarl opened the pull request:
Adds pump for Segment.com

lonelycode commented on pumps/segment.go:
```diff
@@ -0,0 +1,99 @@
+
+	for i, v := range data {
+		analyticsRecord := AnalyticsRecord{}
+		err := msgpack.Unmarshal(v.([]byte), &analyticsRecord)
```
Does this actually work?? You shouldn't need to unmarshal the data using msgpack, it's already unmarshalled when it hits this handler. There is a special case for uptime data, which issn't handled in other pumps. See the [influx handler](https://github.com/TykTechnologies/tyk-pump/blob/develop/pumps/influx.go#L92) for proof ;-)

JosephEarl replied:
I copied this over from some code I had against the 1.9 pump before it was separated out, so may not be needed anymore.

You are saying I can just do 
```
analyticsRecord := v.(AnalyticsRecord)
```
?

lonelycode replied:
Yup, no need to try and decode (that would generate an error I think).

JosephEarl commented:
Removed the JSON decoding

lonelycode commented:
Stupid question, but I gotta ask - but have you successfully used this with segment?

JosephEarl commented:
I'm still doing some testing with Segment so I'd wait a bit before merging just to confirm it's OK.

JosephEarl commented:
Updated and added a test (ignored currently - but you can enter your own Segment key and remove the Skip to try). Here's a screenshot of an event in the Segment debugger:

![segmenttrackhit](https://cloud.githubusercontent.com/assets/2513147/14510941/05c2a352-01cc-11e6-8e68-deae2483dfab.png)

lonelycode commented:
Amazing! Will merge, should be available in the nightlies in about an hour (will publish URLs in the community forum soon)

JosephEarl commented:
Great!
